Shooting.py

```diff
@@ -0,0 +1,112 @@
+from Utils import distance, HexToTuple
+import GameData
+
+resultDict = dict[str, any]  # result dictionary type
+
+
+def neutralityCheck(opponentID: int) -> bool:
+    """
+    Checks if we can attack opponent with specified id
+    The player whose turn it is can attack this
+    opponent only if this opponent attacked him on their previous
+    turn or if this opponent was not attacked by the third player on their previous turn.
+    :param opponentID: id of opponent we want to attack
+    :return: True if we can attack an opponent, False otherwise
+    """
+
+    # attack_matrix is dictionary where: key is playerID, value is list of IDs of opponents he attacked
+    for pid, attacks in GameData.gameState["attack_matrix"].items():
```

Comment:
I my opinion using pid name here is a bit of confusing. Usually pid name is used in the context of processes (process ID) or threads. You are actually using threads here so pid in my opinion is not the best choice.